scaffolding-inspec/lib/scaffolding.sh

```diff
@@ -0,0 +1,113 @@
+#
+# A scaffolding for InSpec
+#
+
+scaffolding_load() {
+  : "${scaffold_inspec:=chef/inspec}"
+
+  pkg_deps=(
+    "${pkg_deps[@]}"
+    "$scaffold_inspec"
+    "core/cacerts"
+  )
+  pkg_build_deps=(
+    "${pkg_build_deps[@]}"
+  )
+
+  pkg_svc_user="root"
+  pkg_svc_run="set_just_so_you_will_render"
+}
+
+do_default_download() {
+  return 0
+}
+
+do_default_verify() {
+  return 0
+}
+
+do_default_unpack() {
+  return 0
+}
+
+do_default_build_service() {
+  ## Create hooks
+  mkdir -p "$pkg_prefix/hooks"
+  chmod 0750 "$pkg_prefix/hooks"
+
+  # Run hook
+  cat << EOF >> "$pkg_prefix/hooks/run"
+#/bin/sh
+
+export SSL_CERT_FILE="{{pkgPathFor "core/cacerts"}}/ssl/cert.pem"
+
+# InSpec will try to create a .cache directory in the user's home directory
+# so this needs to be someplace writeable by the user
+export HOME={{pkg.svc_var_path}}
+
+exec 2>&1
+
+while true; do
+{{pkgPathFor "chef/inspec"}}/bin/inspec exec {{#each cfg.profiles as |profile| ~}} {{profile}} {{/each ~}} --json-config {{pkg.svc_config_path}}/inspec.json --target-id {{sys.member_id}}
+sleep {{cfg.interval}}
+
+done
+EOF
+  chmod 0750 "$pkg_prefix/hooks/run"
+}
+
+do_default_build() {
+  return 0
+}
+
+do_default_install() {
+  mkdir -p "$pkg_prefix/config"
+  chmod 0750 "$pkg_prefix/config"
+  cat << EOF >> "$pkg_prefix/config/inspec.json"
+{
+  "reporter": {
+      "automate" : {
+          "stdout" : "{{cfg.reporter.stdout}}",
+          "url" : "{{cfg.reporter.url}}",
+          "token" : "{{cfg.reporter.token}}",
+          "node_name" : "{{ sys.hostname }}",
+          "verify_ssl" : false
```

Comment:
You need the `node_uuid` here or it won't report to Automate.
```suggestion
          "verify_ssl" : false,
          "node_uuid": {{sys.member_id}}
```

Reply:
We're adding `node_uuid` to the command line arguement, but I don't see how adding it to the json will hurt, so I'll add it there too.

Reply:
#why_not_both?